ctapipe/ml/tools/train_energy_regressor.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+
+from ctapipe.core import Tool
+from ctapipe.core.traits import Int, Path
+from ctapipe.io import TableLoader
+from ctapipe.ml.apply import CrossValidator, EnergyRegressor
+
+from ..preprocessing import check_valid_rows
+from ..sklearn import Regressor
+
+
+class TrainEnergyRegressor(Tool):
+    """Train a ML model for energy regression."""
+
+    name = "ctapipe-train-regressor"
+    description = __doc__
+
+    output_path = Path(
+        default_value=None,
+        allow_none=False,
+        directory_ok=False,
+    ).tag(config=True)
+
+    n_events = Int(default_value=None, allow_none=True).tag(config=True)
+    random_seed = Int(default_value=0).tag(config=True)
+
+    aliases = {
+        ("i", "input"): "TableLoader.input_url",
+        ("o", "output"): "TrainEnergyRegressor.output_path",
```

Comment:
I'd suggest adding an alias for `CrossValidator.output_path` since that is used often (and in the tests). 

Related to that: I added this field to my ml_config.yaml under the EnergyRegressor section, and it did not complain, but also did not write out a file.

Reply:
For more info: I ran:
```sh
ctapipe-ml-train-regressor --config ml_config.yaml --input gamma_diffuse_dl2_train_small.dl2.h5 --output energy_regressor.pkl --log-level=INFO
```

And in the ml_config.yaml file, I have this inside the TrainEnergyRegressor: / EnergyRegressor: section
```yaml
...    
    FeatureGenerator:
      features:
        - ["area", "hillas_width * hillas_length"]

    CrossValidator:
      n_cross_validations: 5
      output_path: "energy_cross_validation.h5"  # this is ignored (perhaps also all?) 
```

No output is written by CrossValidate, however when I add `--CrossValidator.output_path="something.h5"` I get an output!.  So something is perhaps wrong with the config system or how the config file is being parsed

Reply:
The config I get in the provanance log confirms this (and also: why does it include an EnergyRegressor?)
```
1098   │             "CrossValidator": {
1099   │                "n_cross_validations": 5,
1100   │                "output_path": null,
1101   │                "rng_seed": 1337,
1102   │                "EnergyRegressor": {
1103   │                   "model": null,
1104   │                   "QualityQuery": {
1105   │                      "quality_criteria": [
1106   │                         [
1107   │                            "HillasValid",
1108   │                            "HillasReconstructor_is_valid"
1109   │                         ],
1110   │                         [
1111   │                            "enough intensity",
1112   │                            "hillas_intensity > 50"
1113   │                         ],
1114   │                         [
1115   │                            "Positive width",
1116   │                            "hillas_width > 0"
1117   │                         ],
1118   │                         [
1119   │                            "enough pixels",
1120   │                            "morphology_n_pixels > 3"
1121   │                         ],
1122   │                         [
1123   │                            "not clipped",
1124   │                            "leakage_intensity_width_2 < 0.5"
1125   │                         ]
1126   │                      ]
1127   │                   },
1128   │                   "FeatureGenerator": {
1129   │                      "features": [
1130   │                         [
1131   │                            "area",
1132   │                            "hillas_width * hillas_length"
1133   │                         ]
1134   │                      ]
1135   │                   }
1136   │                }
1137   │             }
1138   │          }
1139   │       },
1140   │       "status": "completed",
1141   │       "duration_min": 0.07745000000008773
1142   │    }
```

Reply:
@kosack The ml config example has a wrong indentation for the `CrossvalidatorConfig`. It must be a child of `TrainEnergyRegressor`, not of `EnergyRegressor`